### PR TITLE
MULE-12545: test connection support in smart connectors

### PR DESCRIPTION
**Pull request by fernandezlautaro**

### CODE REVIEW

File: modules/extensions-xml-support/src/test/resources/modules/nested/module-global-element-another-proxy.xml [hunk 1]
```diff
@@ -13,7 +13,9 @@
     <property name="somePassConfig" type="string" defaultValue="some-password"/>
     <property name="port" type="string"/>
 
-    <module-global-element-proxy:config name="another-proxy-config" someUserConfig="#[properties.someUserConfig]" somePassConfig="#[properties.somePassConfig]" port="#[properties.port]"/>
+    <module-global-element-proxy:config name="another-proxy-config" someUserConfig="#[properties.someUserConfig]" somePassConfig="#[properties.somePassConfig]" port="#[properties.port]" >
+        <module-global-element-proxy:connection />
```
Comment (pablolagreca) [inline modules/extensions-xml-support/src/test/resources/modules/nested/module-global-element-another-proxy.xml]: Will this element contain all the parameters of the module that are related to the connection?. That would make it consistent with what we have in the Java connectors

File: modules/extensions-xml-support/src/test/java/org/mule/test/functional/ModuleTestConnectionSuccessful.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import org.mule.runtime.api.connection.ConnectionProvider;
+import org.mule.runtime.api.connection.ConnectionValidationResult;
+import org.mule.runtime.api.exception.MuleException;
+import org.mule.runtime.extension.api.runtime.config.ConfigurationInstance;
+import org.mule.test.runner.RunnerDelegateTo;
+
+import java.util.Collection;
+
+@RunnerDelegateTo(Parameterized.class)
+public class ModuleTestConnectionSuccessful extends AbstractXmlExtensionMuleArtifactFunctionalTestCase {
```
Comment (pablolagreca) [inline modules/extensions-xml-support/src/test/java/org/mule/test/functional/ModuleTestConnectionSuccessful.java]: ModuleTestConnectionSuccessful -> ModuleTestConnectionSuccessfulTestCase

File: modules/extensions-xml-support/src/test/java/org/mule/test/functional/ModuleWithMultipleGlobalElementsTestCase.java [hunk 3]
```diff
@@ -58,6 +62,27 @@ protected String getConfigFile() {
   }
 
   @Test
+  public void testConnectionconfigPatternA() throws Exception {
```
Comment (pablolagreca) [inline modules/extensions-xml-support/src/test/java/org/mule/test/functional/ModuleWithMultipleGlobalElementsTestCase.java]: testConnectionconfigPatternA -> testConnectionConfigPatternA

File: modules/extensions-xml-support/src/test/java/org/mule/test/functional/ModuleWithMultipleGlobalElementsTestCase.java [hunk 4]
```diff
@@ -58,6 +62,27 @@ protected String getConfigFile() {
...
+    doTestConnection("configPatternA");
+  }
+
+  @Test
+  public void testConnectionconfigPatternB() throws Exception {
```
Comment (pablolagreca) [inline modules/extensions-xml-support/src/test/java/org/mule/test/functional/ModuleWithMultipleGlobalElementsTestCase.java]: testConnectionconfigPatternB -> testConnectionConfigPatternB

File: modules/spring-config/src/main/java/org/mule/runtime/config/spring/internal/dsl/model/extension/xml/MacroExpansionModuleModel.java [hunk 5]
```diff
@@ -67,6 +68,18 @@
    */
   public static final String TNS_PREFIX = "tns";
 
+  /**
+   * literal that represents the name of the connection element for any given module. If the module's name is math, then the value of
```
Comment (pablolagreca) [inline modules/spring-config/src/main/java/org/mule/runtime/config/spring/internal/dsl/model/extension/xml/MacroExpansionModuleModel.java]: math?

**Comment by pablolagreca** [comment]
I didn't find a functional test or how is supported in the code to do test connectivity over a module with two connection global elements

File: modules/extensions-xml-support/src/test/resources/modules/nested/module-global-element-another-proxy.xml (hunk 1 above)
Comment (fernandezlautaro) [reply modules/extensions-xml-support/src/test/resources/modules/nested/module-global-element-another-proxy.xml]: left for other JIRA https://www.mulesoft.org/jira/browse/MULE-13409

**Pull request merged**